docs/source/troubleshooting/tracing.rst

```diff
@@ -55,33 +55,101 @@ Below is the typical sequence to start an *event trace* from an SSH/serial termi
 
    Use this with caution and sparingly. Writing to the trace buffer adds additional latency.
 
-Analyzing the results
+#. Stop tracing.
+
+   Trace events are written to a circular buffer, so newer trace contents overwrite older trace contents. Stop tracing as soon as possible after the event of interest to ensure important trace information is not lost.
+
+   .. code:: bash
+
+      echo 0 > /sys/kernel/debug/tracing/tracing_on
+
+Extracting the trace
 =====================
 
-The resulting trace can be extracted with ``trace-cmd`` which can be installed from packages feeds with
+The resulting trace can be extracted to a binary file with ``trace-cmd`` which can be installed from packages feeds with
 
 .. code:: bash
 
    opkg install trace-cmd
 
-To extract the trace, use
+To extract the trace to the default filename ``trace.dat``, use
 
 .. code:: bash
 
    trace-cmd extract
 
-The resulting trace file ``trace.dat`` can then be loaded in the front-end GUI tool ``kernelshark`` for off-target analysis.
+Alternatively, the trace can instead be extracted in a human readable textual representation using the following command. Extracting the trace can be a slow operation if it is large. 
+
+.. code:: bash
+
+   cat /sys/kernel/debug/tracing/trace > trace.txt
+
+Analyzing the results
+=====================
+
+A resulting binary trace file can then be loaded in the front-end GUI tool Kernelshark for off-target analysis.
 
    .. image:: resources/trace-loaded-in-kernelshark.png
 
-``kernelshark`` can be used on a Linux desktop distribution or on Windows with Windows Subsystem for Linux (WSL).
+Kernelshark can be used on a Linux desktop distribution or on Windows using the Windows Subsystem for Linux (WSL).
+
+Kernelshark using Windows Subsystem for Linux (WSL)
+---------------------------------------------------
+
+Kernelshark is a Linux application. You can run it on a native Linux desktop installation or a virtual machine installation. If you would like to run Kernelshark from within a Windows desktop environment, you can do so using the Windows Subsystem for Linux (WSL).
+
+A complete guide to installing and using WSL is outside the scope of this document. However, the following steps show the general process to install and use Kernelshark. For more information, see the `Microsoft documentation <https://learn.microsoft.com/en-us/windows/wsl/install>`_.
+
+#. Install WSL
+
+   From Powershell or Command Prompt run in administrator mode:
+
+   .. code:: console
```

Comment:
We should probably use `batch` for windows commands. Ref https://pygments.org/languages/ used by sphinx.

Reply:
Done.